tests/integration/backup_tests/test_sharding_backups.py

```diff
@@ -0,0 +1,437 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
```

Comment:
copied with minor modifications from VM charm (with @Gu1nness recent improvements to tests)